File: src/days/Day5.java
package days;

import setup.Day;
import util.Line;

import java.util.Arrays;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Day5 extends Day {
    private List<Line> lines;

    @Override
    public void processInput() {
        String[] lines = input.split("\n");
        this.lines = Arrays.stream(lines).map(Line::toLine).toList();
    }

    @Override
    public Object part1() {
        Set<Line.Point> overlap = new HashSet<>();
        for (int i = 0; i < lines.size(); i++) {
            Line lineA = lines.get(i);
            if (!lineA.horizontal() && !lineA.vertical()) continue;
            for (int j = i + 1; j < lines.size(); j++) {
                Line lineB = lines.get(j);
                if (!lineB.horizontal() && !lineB.vertical()) continue;
                overlap.addAll(lineA.overlap(lineB));
            }
        }

        return overlap.size();
    }

    @Override
    public Object part2() {
        return null;
    }

    @Override
    public int getDay() {
        return 5;
    }

    @Override
    public boolean isTest() {
        return false;
    }

}

File: src/days/day5.py
from src.setup.day import Day
from src.util.solution import solution

class Day5(Day):
    @property
    def day(self):
        return 5

    @solution("623")
    def part1(self) -> object:
        fresh_ranges, ingredients = self.input
        fresh_ranges = [tuple(map(int, x.split("-"))) for x in fresh_ranges]
        fresh_ranges = sorted(fresh_ranges)
        ingredients = [int(x) for x in ingredients]
        # print(fresh_ranges, ingredients)

        fresh = 0
        for ingredient in ingredients:
            for low, high in fresh_ranges:
                if low > ingredient:
                    break
                if low <= ingredient <= high:
                    fresh += 1
                    break
        return fresh

    @solution("")
    def part2(self) -> object:
        return None

    def read_input(self):
        super().read_input()
        return [x.split("\n") for x in self.raw_input.split("\n\n")]



# Day5("test").run()
Day5().run()